docs/uds/scan_modes.md

```diff
@@ -99,6 +99,22 @@ The discovery procedure is dependend on the used addressing scheme.
 From a high level perspective, the destination id is iterated and a valid payload is sent.
 If a valid answer is received, an ECU has been found.
 
+## Session Scan
+
+UDS has the concept of sessions.
+Different sessions can for example offer different services.
+A session is identified by a 1 byte session ID.
+The UDS standard defines a set of well known session IDs, but vendors are free to add their own sessions.
+Some sessions might only be available from a specific ECU state (e.g. current session, enabled/configured ECU features, coding, ...).
+Most of those preconditions cannot be detected automatically and might require vendor specific knowledge.
+
+This scanner tries to find all available session transitions.
+Starting from the default session (0x01), it iterates all session IDs and tries to access this session.
+If the ECU replays with a positive response, the session is available, else the session is not available (at leased from the current state).
```

Comment:
```suggestion
If the ECU replies with a positive response, the session is available.
In case of a negative response, the session is considered not available from state.
```